src/katgpucbf/queue_item.py

```diff
@@ -42,37 +42,51 @@ class QueueItem:
 
     #: Timestamp of the item.
     timestamp: int
-    #: A list of GPU event markers generated by an
+    #: The latest GPU event marker per
     #: :class:`~katsdpsigproc.abc.AbstractCommandQueue`.
-    events: list[AbstractEvent]
+    _events: dict[AbstractCommandQueue, AbstractEvent]
 
     def __init__(self, timestamp: int = 0) -> None:
         self.reset(timestamp)
+        self._events = {}
 
-    def add_marker(self, command_queue: AbstractCommandQueue) -> None:
+    def add_marker(self, command_queue: AbstractCommandQueue) -> AbstractEvent:
         """Add an event to the list of events in the QueueItem.
 
         The event represents all previous work enqueued to `command_queue`.
         """
-        self.events.append(command_queue.enqueue_marker())
+        marker = command_queue.enqueue_marker()
+        self._events[command_queue] = marker
+        return marker
```

Comment:
1. Not to be too pedantic, but does the docstring need to be updated? Specifically around "Add an event to the *list of events*.
    * Admittedly I don't have a suggestion, but technically we're not working with lists anymore.
    * Feel free to ignore this request.
2. What happens if the caller doesn't do anything with the returned `marker`?
    * e.g. fgpu/engine.py Lines [667](https://github.com/ska-sa/katgpucbf/blob/NGC-1042-reset-events/src/katgpucbf/fgpu/engine.py#L667), [720](https://github.com/ska-sa/katgpucbf/blob/NGC-1042-reset-events/src/katgpucbf/fgpu/engine.py#L720) and [850](https://github.com/ska-sa/katgpucbf/blob/NGC-1042-reset-events/src/katgpucbf/fgpu/engine.py#L850) (few more examples in xbgpu/engine.py as well)

Reply:
> 1. Not to be too pedantic, but does the docstring need to be updated? Specifically around "Add an event to the _list of events_.

Fair point - @james-smith-za any suggestions on how to reword it, that avoids saying anything about the internal representation?

>     2. What happens if the caller doesn't do anything with the returned `marker`?

That's perfectly fine - it's just returned in case the caller wants to know (there was one place where I needed it).

Reply:
Something like "Add an (awaitable possibly?) event to the QueueItem" would probably be fine IMO.